### ISSUE TITLE
zlib issue: External LLVMFunction rb_lastline_set cannot be found

### ISSUE THREAD

**Issue by gogainda** [oracle/truffleruby#2170]
```
 1) Error:
TestZlibGzipFile#test_lineno:
RuntimeError: External LLVMFunction rb_lastline_set cannot be found. (com.oracle.truffle.llvm.runtime.except.LLVMLinkerException)
Translated to internal error
    /Users/novoi/tmp/zlib/ext/zlib/zlib.c:4324:in `rb_gzreader_gets'
    /Users/novoi/.rubies/truffleruby-dev/lib/truffle/truffle/cext_ruby.rb:39:in `gets'
    /Users/novoi/tmp/zlib/test/zlib/test_zlib.rb:644:in `block (2 levels) in test_lineno'
    /Users/novoi/.rubies/truffleruby-dev/lib/truffle/truffle/cext.rb:1003:in `rb_yield'
    call.c:70:in `rb_yield'
    /Users/novoi/.rubies/truffleruby-dev/lib/truffle/truffle/cext.rb:1462:in `rb_ensure'
    exception.c:68:in `rb_ensure'
    /Users/novoi/tmp/zlib/ext/zlib/zlib.c:3131:in `gzfile_wrap'
    /Users/novoi/tmp/zlib/ext/zlib/zlib.c:3173:in `gzfile_s_open'
    /Users/novoi/tmp/zlib/ext/zlib/zlib.c:3792:in `rb_gzreader_s_open'
    /Users/novoi/.rubies/truffleruby-dev/lib/truffle/truffle/cext_ruby.rb:39:in `open'
    /Users/novoi/tmp/zlib/test/zlib/test_zlib.rb:643:in `block in test_lineno'
    /Users/novoi/.rubies/truffleruby-dev/lib/mri/tempfile.rb:332:in `create'
    /Users/novoi/tmp/zlib/test/zlib/test_zlib.rb:639:in `test_lineno'
```
To reproduce:
```
git clone git@github.com:ruby/zlib.git 
cd zlib
bundle install
bundle exec rake
```

**Comment by eregon** [comment]
Is TruffleRuby tested in https://github.com/ruby/zlib or is there a PR for it? Where does this run come from?

Anyway, good thing to fix, and we should be able to untag the corresponding MRI test.
@aardvark179 Can you take care of this one?

**Comment by gogainda** [comment]
zlib is not tested against truffleruby, I just run it locally

**Comment by gogainda** [comment]
There is a good discussion how to decrease MEM usage: https://github.com/ruby/zlib/issues/19#issuecomment-721615795

**Comment by eregon** [comment]
rb_lastline_set implemented in b46b9b79ae7a06b6e5bfb2f974ad6191f07b6e45 by @aardvark179.
